crates/task-impls/src/da.rs

```diff
@@ -108,35 +108,31 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, V: Versions> DaTaskState<TYP
                 // the `DaProposalRecv` event. Otherwise, the view number subtraction below will
                 // cause an overflow error.
                 // TODO ED Come back to this - we probably don't need this, but we should also never receive a DAC where this fails, investigate block ready so it doesn't make one for the genesis block
```

Comment:
Can we resolve this TODO comment?  I don't think receiving a proposal has much to do with the block ready event.  A bad leader could send us proposals for old views so we need this check to discard them early.

Longer term follow up, what if we get a valid DAProposal late, is there any value in backfilling the leaf with the full block data?